nexus/db-queries/src/db/datastore/volume.rs

```diff
@@ -2160,6 +2469,71 @@ fn replace_region_in_vcr(
     Ok(new_vcr)
 }
 
+/// Replace a read-only target in a VolumeConstructionRequest
+///
+/// Note that UUIDs are not randomized by this step: Crucible will reject a
+/// `target_replace` call if the replacement VolumeConstructionRequest does not
+/// exactly match the original, except for a single Region difference.
+///
+/// Note that the generation number _is not_ bumped in this step.
+fn replace_read_only_target_in_vcr(
+    vcr: &VolumeConstructionRequest,
+    old_target: SocketAddrV6,
+    new_target: SocketAddrV6,
+) -> anyhow::Result<(VolumeConstructionRequest, usize)> {
+    let mut new_vcr = vcr.clone();
+
+    let mut parts: VecDeque<&mut VolumeConstructionRequest> = VecDeque::new();
+    parts.push_back(&mut new_vcr);
+
+    let mut replacements = 0;
+
+    let mut old_target_found = false;
+
+    while let Some(vcr_part) = parts.pop_front() {
+        match vcr_part {
+            VolumeConstructionRequest::Volume {
+                sub_volumes,
+                read_only_parent,
+                ..
+            } => {
+                for sub_volume in sub_volumes {
+                    parts.push_back(sub_volume);
```

Comment:
This will replace targets in a `sub_volume` as well?
If so, this function's name is misleading.

Reply:
addressed in 18837e338